lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -368,13 +365,25 @@ async sub transform_transaction {
             timestamp    => $int_timestamp,
         );
 
-        my @transactions = await $self->_check_contract_transaction($transaction, $receipt);
-        push @transactions, $transaction if ($amount->bgt(0));
-
         if (!$self->accounts() || ($self->latest_accounts_update + UPDATE_ACCOUNTS <= time)) {
             await $self->get_hash_accounts();
         }
 
+        # if the gas is empty we don't proceed
+        unless ($gas && $decoded_transaction->{gasPrice}) {
+            my $our_transaction = await $self->_set_transaction_type($transaction);
+            warn sprintf("%s: Failed to get Receipt & GasPrice for transaction: %s", $self->currency_symbol, $decoded_transaction->{hash})
+                if $our_transaction;
```

Comment:
@arshad-binary if in case we receive the value in `$our_transaction` from sub _set_transaction_type, we will not display the warn log even though gas was empty.

Reply:
As per discussion we need to check that txn belongs to us.